storage/gcloud/aio/storage/storage.py

```diff
@@ -78,32 +71,50 @@ def __init__(self, project, service_file, token=None, session=None):
         resp.raise_for_status()
         return await resp.json()
 
-    async def upload(self, bucket, object_name, file_data, headers=None,
-                     session=None):
-        # pylint: disable=too-many-arguments
-        # https://cloud.google.com/storage/docs/json_api/v1/how-tos/simple-upload
+    async def upload(self, bucket: str, object_name: str,
+                     file_data, content_type: str,
+                     headers=None, session: aiohttp.ClientSession = None,
+                     timeout: int = 120, upload_type_resumable_always: bool = False):
```

Comment:
[As described in #41](https://github.com/talkiq/gcloud-aio/issues/41#issuecomment-440078798), I'd love for this variable to be an `Optional[bool]` with the following behaviour:
- `None` (default) -> use resumable uploads if filesize exceeds 5MB
- `True` -> always use resumable uploads
- `False` -> never use resumable uploads. Yes, this would imply that the Google response would error if file sizes ever exceed 5MB (but that is the library user's choice!).

I think a slightly more comprehensible variable name here would be `force_resumable_upload`.